doc/book/in-depth-guide/first-module.md

```diff
@@ -332,11 +332,11 @@ return [
 
 The above configuration tells the application that the folder
 `module/Blog/view/` has view files in it that match the above described default
-scheme. It is important to note that with this you can not only ship view files
+scheme [[Not clear to me what default scheme is being referred to here. Do you mean the path structure? Could it be stated explicitly?]] . It is important to note that with this you can not only ship view files
```

Comment:
We reference the "above described default scheme" — it's detailed two paragraphs prior as `view/{namespace}/{controller}/{action}.phtml`. In fact, it's the paragraph immediately prior to the cut in this diff!

I guess what I'm wondering is: should we NOT expect the reader to remember this information by this point? If this is the case, do you have any concrete suggestions that would help readers remember this fact?

Reply:
I don't think it's so much a matter of the reader remembering as understanding what's being referred to. I worked it out, but I had to read it a few times (might just be me :-) I would rephrase it as:

"The above configuration tells the application that the folder ```module/Blog/view/``` has view files in it that match the standard path format: ```view/{namespace}/{controller}/{action}.phtml```.

[As a side note, I have just started my first developer role, and while I have a good grounding in PHP, frameworks and ZF are new to me. My overall impression of the tutorial is that in some places things are spelt out very clearly and in others a level of knowledge beyond framework/ZF newbie is assumed and makes it hard to follow. For example, the next sentence

"It is important to note that with this you can not only ship view files for your module, but you can also overwrite view files from other modules."

isn't at all clear to me. Maybe the tutorial is aimed at people with more PHP/coding/framework experience than I have, but I don't know what "ship view files for your module" means here, or why you would want to overwrite view files from other modules.].

Reply:
I like the rephrasing! Let's go with that, please!

Regarding this:

> It is important to note that with this you can not only ship view files for your module, but you can also overwrite view files from other modules.

What this means is that, assuming another module uses the standard path format for view scripts, it's possible for you to provide substitutions for those view scripts within your own module by mimicing that structure, and providing configuration for it.

As an example, let's say we have another module, "News", and it has a controller "Updates" with the action "Breaking". Assuming the module follows the standard layout, it would have a view script in `view/news/updates/breaking.phtml`, and configuration like the following:

```php
'view_manager' => [
    'template_path_stack' => [
        'news' => __DIR__ . '/../view',
    ],
],
```

In our own `Blog` module, we could provide configuration for that alongside our own view script configuration:

```php
'view_manager' => [
    'template_path_stack' => [
        'blog' => __DIR__ . '/../view',
        'news' => __DIR__ . '/../view/news',
    ],
],
```

The above tells the view manager to look in our own `view/news/` subdirectory for matching templates.

If our `Blog` module is registered later than the `News` module, that means the view manager will scan the `Blog` module's `view/news/` subdirectory for matching templates _before_ scanning the `News` module — allowing us to selectively override view scripts.

I'm not sure if this is too much information to introduce at this point in the tutorial, though. Thoughts?

Reply:
Updated the patch with the rephrasing.

Thanks for this detailed explanation. It's very clear, although I haven't learnt enough about ZF yet to fully understand how this would work. I think inserting this info would break the flow of the tutorial so I would keep the note as it is at the moment, with a slight clarification (if I understand it correctly):

"It is important to note that the view_manager configuration not only allows you to ship view files for your module, but also to overwrite view files from other modules."

and make the last six words a link to a separate topic on the subject with the same name. Maybe put it in a tutorial Appendix that could hold more detailed explanations of things that are important, but would break the flow of the tutorial?